shared/params.go

```diff
@@ -201,17 +201,29 @@ func ParseSHAParam(r *http.Request) (runSHA string, err error) {
 	return sha[:10], nil
 }
 
+// ParseSHA validates the given 'sha' value, cropping it to 10 chars.
+// It returns "latest" by default. (and in error cases).
+func ParseSHA(sha string) (runSHA string, err error) {
+	sha, err = ParseSHAFull(sha)
+	if err != nil || !SHARegex.MatchString(sha) {
+		return sha, err
+	}
+	return sha[:10], nil
+}
+
 // ParseSHAParamFull parses and validates the 'sha' param for the request.
 // It returns "latest" by default (and in error cases).
 func ParseSHAParamFull(r *http.Request) (runSHA string, err error) {
 	// Get the SHA for the run being loaded (the first part of the path.)
 	runSHA = "latest"
```

Comment:
This variable seems unused now in `ParseSHAParamFull`?

Reply:
Done.